pkg/server/batch_check_test.go

```diff
@@ -0,0 +1,343 @@
+				Message: condition.ErrEvaluationFailed.Error(),
+			},
+		},
+		`test_throttled_error`: {
+			inputError: &commands.ThrottledError{Cause: errors.New(errMsg)},
+			expectedOutput: &openfgav1.CheckError{
+				Code:    &openfgav1.CheckError_InputError{InputError: openfgav1.ErrorCode_validation_error},
+				Message: errMsg,
+			},
+		},
+		`test_deadline_exceeded`: {
+			inputError: context.DeadlineExceeded,
+			expectedOutput: &openfgav1.CheckError{
+				Code:    &openfgav1.CheckError_InternalError{InternalError: openfgav1.InternalErrorCode_deadline_exceeded},
+				Message: context.DeadlineExceeded.Error(),
+			},
+		},
+		`test_generic_error`: {
+			inputError: errors.New(errMsg),
+			expectedOutput: &openfgav1.CheckError{
+				Code:    &openfgav1.CheckError_InternalError{InternalError: openfgav1.InternalErrorCode_internal_error},
+				Message: errMsg,
+			},
+		},
+	}
+
+	for name, scenario := range scenarios {
+		t.Run(name, func(t *testing.T) {
+			result := transformCheckCommandErrorToBatchCheckError(scenario.inputError)
+			require.Equal(t, scenario.expectedOutput, result)
+		})
+	}
+}
+
+func TestTransformCheckResultToProto(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+
+	var expectedResult = map[string]*openfgav1.BatchCheckSingleResult{}
+	outcomes := map[commands.CorrelationID]*commands.BatchCheckOutcome{}
+
+	// Create two fake outcomes, one happy path and one error
+	// These two make 100% coverage
+	id1 := commands.CorrelationID("abc123")
+	outcomes[id1] = &commands.BatchCheckOutcome{
+		CheckResponse: &graph.ResolveCheckResponse{Allowed: true},
+	}
+
+	id2 := commands.CorrelationID("def456")
+	outcomes[id2] = &commands.BatchCheckOutcome{Err: graph.ErrResolutionDepthExceeded}
+
+	// format the expected final output of the transform function
+	expectedResult[string(id1)] = &openfgav1.BatchCheckSingleResult{
+		CheckResult: &openfgav1.BatchCheckSingleResult_Allowed{Allowed: true},
+	}
+
+	expectedResult[string(id2)] = &openfgav1.BatchCheckSingleResult{
+		CheckResult: &openfgav1.BatchCheckSingleResult_Error{
+			Error: &openfgav1.CheckError{
+				Code:    &openfgav1.CheckError_InputError{InputError: openfgav1.ErrorCode_authorization_model_resolution_too_complex},
+				Message: graph.ErrResolutionDepthExceeded.Error(),
+			},
+		},
+	}
+
+	t.Run("test_transform_check_result_to_proto", func(t *testing.T) {
+		result := transformCheckResultToProto(outcomes)
+		require.Equal(t, expectedResult, result)
+	})
+}
+
+func BenchmarkTransformCheckResultToProto(b *testing.B) {
+	outcomes := map[commands.CorrelationID]*commands.BatchCheckOutcome{
+		"abc123": {
+			CheckResponse: &graph.ResolveCheckResponse{
+				Allowed: true,
+			},
+		},
+		"def456": {
+			Err: graph.ErrResolutionDepthExceeded,
+		},
+	}
+
+	b.Run("benchmark_transform_check_result_to_proto", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			transformCheckResultToProto(outcomes)
```

Comment:
IMO, `transformCheckResultToProto` is not complex enough to warrant a benchmark, and it's not in the critical path.

A more useful benchmark could be testing what happens when the batch check contains `MaxChecksPerBatchCheck` to ensure that we don't introduce regressions when it comes to parallelizing the work.

Reply:
AFAICT, it _is_ in the critical path which is why I asked for this benchmark test to begin with.

Reply:
Ah my bad, "critical path" isn't the right term...

Reply:
I ended up leaving this benchmark in and adding an explainer comment to the test. It actually is part of the critical path, transforming all final check results into the BatchCheck proto format after they've been gathered.